frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { OnSelect } from '@patternfly/react-table';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export const useSelectList = <R extends K8sResourceCommon>(
+  data: R[],
+  visibleRows: string[],
+  onRowSelected: (rows: R[]) => void,
+): {
+  onSelect: OnSelect;
+  selectedRows: string[];
```

Comment:
@gnehapk @rawagner I believe this logic will be much simpler(we will be able to remove use of uniq) if we use sets instead of this. WDYT?

Reply:
+1 for sets